packages/protocol/thread/thread.ts

```diff
@@ -296,17 +297,34 @@ class _ThreadFront {
     await client.Session.ensureProcessed({ level }, sessionId);
   }
 
-  async listenForLoadChanges(listenerCallback: (params: loadedRegions) => void) {
-    // This is a placeholder which logs loading changes to the console.
-    const sessionId = await this.waitForSession();
+  private _listeningForLoadChanges: boolean = false;
+  private _loadedRegionsListeners: LoadedRegionListener[] = [];
+  private _mostRecentLoadedRegions: LoadedRegions | null = null;
+
+  async listenForLoadChanges(listener: LoadedRegionListener) {
+    this._loadedRegionsListeners.push(listener);
+
+    if (!this._listeningForLoadChanges) {
+      this._listeningForLoadChanges = true;
+
+      const sessionId = await this.waitForSession();
+
+      client.Session.addLoadedRegionsListener((loadedRegions: LoadedRegions) => {
+        this._mostRecentLoadedRegions = loadedRegions;
+
+        if (loadedRegions.loading) {
```

Comment:
I realize this is existing logic, but just to mention it...I think this is always `true`, since `loading` is just an array.

Reply:
Ooo ok good point! I'll clean that up too then while I'm in here.